spyder/app/mainwindow.py

```diff
@@ -1975,6 +1977,11 @@ def main(options, args):
             pass
     CONF.set('main', 'previous_crash', previous_crash)
 
+    if options.offline:
+        CONF.set('ipython_console', 'info_widget', False)
+        CONF.set('help', 'enable', False)
+        CONF.set('onlinehelp', 'enable', False)

```

Comment:
There's a missing `else` here to set these options to `True`. Else, these elements would be hidden until users reset their preferences.

Reply:
Yes this was something I was unsure about.   Because of how the configuration files are parsed, if you set these configuration options,  the spyder.ini values are ignored and overwritten.  

So this could cause confusion if a user sets options in spyder.ini and sees them ineffective and overwritten at runtime.  For example if you set help.enable = False in spyder.ini but then run without '--no-web-widgets', if we add an `else: #set to true` phrase your spyder.ini settings won't be honored, without the else it honors the command line, then the spyder.ini.  But in both cases it'll overwrite your spyder.ini values with true or false depending on the flag setting.  

It seems to me that the best thing would be to have the flag overwrite these settings but not write to spyder.ini.  So that to disable these features more permanently you'd need to edit spyder.ini, but disable them for a single run you'd just need to use the flag.  I couldn't find if it was possible to make that happen.

In any case, I was hoping you guys had a canonical way of dealing with this type of issue so whatever you think is best is what it should do.